diff --git a/src/GvimExt/gvimext.cpp b/src/GvimExt/gvimext.cpp
index b682b74..c9ce8eb 100644
--- a/src/GvimExt/gvimext.cpp
+++ b/src/GvimExt/gvimext.cpp
@@ -335,7 +335,7 @@ dyn_gettext_load(void)
 	len = (DWORD)strlen(szBuff);
 	if (dyn_libintl_init(szBuff))
 	{
-	    strcpy(szBuff + len, "lang");
+	    strcpy(szBuff + len, "vim80/lang");
 
 	    (*dyn_libintl_bindtextdomain)(VIMPACKAGE, szBuff);
 	    (*dyn_libintl_textdomain)(VIMPACKAGE);
