diff --git a/src/edit.c b/src/edit.c
index 953b1fd..333f37b 100644
--- a/src/edit.c
+++ b/src/edit.c
@@ -2826,7 +2826,6 @@ ins_compl_upd_pum(void)
     if (compl_match_array != NULL)
     {
 	h = curwin->w_cline_height;
-	update_screen(0);
 	if (h != curwin->w_cline_height)
 	    ins_compl_del_pum();
     }
@@ -2917,9 +2916,6 @@ ins_compl_show_pum(void)
     do_cmdline_cmd((char_u *)"if exists('g:loaded_matchparen')|3match none|endif");
 #endif
 
-    /* Update the screen before drawing the popup menu over it. */
-    update_screen(0);
-
     if (compl_match_array == NULL)
     {
 	/* Need to build the popup menu list. */
@@ -3029,6 +3025,9 @@ ins_compl_show_pum(void)
 	 * part of the screen would be updated.  We do need to redraw here. */
 	dollar_vcol = -1;
 
+	/* Update the screen before drawing the popup menu over it. */
+	update_screen(0);
+
 	/* Compute the screen column of the start of the completed text.
 	 * Use the cursor to get all wrapping and other settings right. */
 	col = curwin->w_cursor.col;
@@ -4848,6 +4847,10 @@ ins_compl_next(
 
 	/* display the updated popup menu */
 	ins_compl_show_pum();
+
+	/* Delete old text to be replaced, since we're still searching and
+	 * don't want to match ourselves!  */
+	ins_compl_delete();
 #ifdef FEAT_GUI
 	if (gui.in_use)
 	{
@@ -4857,10 +4860,6 @@ ins_compl_next(
 	    gui_update_cursor(FALSE, FALSE);
 	}
 #endif
-
-	/* Delete old text to be replaced, since we're still searching and
-	 * don't want to match ourselves!  */
-	ins_compl_delete();
     }
 
     /* Enter will select a match when the match wasn't inserted and the popup
diff --git a/src/popupmnu.c b/src/popupmnu.c
index 4c3ac70..f128971 100644
--- a/src/popupmnu.c
+++ b/src/popupmnu.c
@@ -663,7 +663,7 @@ pum_set_selected(int n, int repeat)
 
 			/* Return cursor to where we were */
 			validate_cursor();
-			redraw_later(SOME_VALID);
+			redraw_win_later(curwin, REDRAW_TOP);
 
 			/* When the preview window was resized we need to
 			 * update the view on the buffer.  Only go back to
@@ -680,7 +680,7 @@ pum_set_selected(int n, int repeat)
 			/* Update the screen before drawing the popup menu.
 			 * Enable updating the status lines. */
 			pum_do_redraw = TRUE;
-			update_screen(0);
+			win_redr_status(curwin);
 			pum_do_redraw = FALSE;
 
 			if (!resized && win_valid(curwin_save))
@@ -693,7 +693,7 @@ pum_set_selected(int n, int repeat)
 			/* May need to update the screen again when there are
 			 * autocommands involved. */
 			pum_do_redraw = TRUE;
-			update_screen(0);
+			win_redr_status(curwin);
 			pum_do_redraw = FALSE;
 		    }
 		}
