diff --git a/src/GvimExt/Make_ming.mak b/src/GvimExt/Make_ming.mak
index 2976ea0..645bca0 100644
--- a/src/GvimExt/Make_ming.mak
+++ b/src/GvimExt/Make_ming.mak
@@ -43,7 +43,7 @@ else
 DEL = del
 endif
 endif
-CXX := $(CROSS_COMPILE)g++
+CXX := ccache $(CROSS_COMPILE)g++
 WINDRES := $(CROSS_COMPILE)windres
 WINDRES_CXX = $(CXX)
 WINDRES_FLAGS = --preprocessor="$(WINDRES_CXX) -E -xc" -DRC_INVOKED
diff --git a/src/Make_cyg_ming.mak b/src/Make_cyg_ming.mak
index f6ab0db..81f7ca6 100644
--- a/src/Make_cyg_ming.mak
+++ b/src/Make_cyg_ming.mak
@@ -402,8 +402,8 @@ MKDIR = mkdir
 DIRSLASH = \\
 endif
 endif
-CC := $(CROSS_COMPILE)gcc
-CXX := $(CROSS_COMPILE)g++
+CC := ccache $(CROSS_COMPILE)gcc
+CXX := ccache $(CROSS_COMPILE)g++
 ifeq ($(UNDER_CYGWIN),yes)
 WINDRES := $(CROSS_COMPILE)windres
 else
@@ -414,7 +414,7 @@ WINDRES_CC = $(CC)
 #>>>>> end of choices
 ###########################################################################
 
-CFLAGS = -Iproto $(DEFINES) -pipe -march=$(ARCH) -Wall
+CFLAGS = -Iproto $(DEFINES) -pipe -march=native -Wall
 WINDRES_FLAGS = --preprocessor="$(WINDRES_CC) -E -xc" -DRC_INVOKED
 EXTRA_LIBS =
 
@@ -578,6 +578,7 @@ else
 ifeq ($(OPTIMIZE), MAXSPEED)
 CFLAGS += -O3
 CFLAGS += -fomit-frame-pointer -freg-struct-return
+LFLAGS += -Wl,-s
 else  # SPEED
 CFLAGS += -O2
 endif
diff --git a/src/Make_ming.mak b/src/Make_ming.mak
index 9ed1a70..659be93 100644
--- a/src/Make_ming.mak
+++ b/src/Make_ming.mak
@@ -11,6 +11,8 @@
 # Last updated by Ken Takata.
 # Last Change: 2014 Oct 21
 
+STATIC_STDCPLUS=yes
+USE_STDCPLUS=yes
 
 # uncomment 'PERL' if you want a perl-enabled version
 #PERL=c:/perl
@@ -33,9 +35,19 @@
 #PYTHON=/home/ron/ActivePython-2.0.0-202/src/Core
 # on my NT box, it's here:
 #PYTHON=c:/python20
+PYTHON_VER=27
+PYTHON=c:/git-sdk-64/mingw64
+PYTHON_HOME=$(PYTHON)
+PYTHONINC=-I$(PYTHON)/include/python2.7
+DYNAMIC_PYTHON_DLL=libpython2.7.dll
 
 # uncomment 'PYTHON3' if you want a python3-enabled version
 #PYTHON3=c:/python31
+PYTHON3_VER=35
+PYTHON3=c:/git-sdk-64/mingw64
+PYTHON3_HOME=$(PYTHON3)
+PYTHON3INC=-I$(PYTHON3)/include/python3.5m
+DYNAMIC_PYTHON3_DLL=libpython3.5m.dll
 
 # uncomment 'TCL' if you want a Tcl-enabled version
 #TCL=c:/tcl
