diff --git a/src/GvimExt/Make_ming.mak b/src/GvimExt/Make_ming.mak
index a6ab3ae..7cbbe43 100644
--- a/src/GvimExt/Make_ming.mak
+++ b/src/GvimExt/Make_ming.mak
@@ -47,7 +47,7 @@ endif
 ifndef WINVER
 WINVER = 0x0501
 endif
-CXX := $(CROSS_COMPILE)g++
+CXX := ccache $(CROSS_COMPILE)g++
 WINDRES := $(CROSS_COMPILE)windres
 WINDRES_CXX = $(CXX)
 WINDRES_FLAGS = --preprocessor="$(WINDRES_CXX) -E -xc" -DRC_INVOKED
diff --git a/src/Make_cyg_ming.mak b/src/Make_cyg_ming.mak
index 62585e9..fc9d5c0 100644
--- a/src/Make_cyg_ming.mak
+++ b/src/Make_cyg_ming.mak
@@ -402,8 +402,8 @@ MKDIR = mkdir
 DIRSLASH = \\
 endif
 endif
-CC := $(CROSS_COMPILE)gcc
-CXX := $(CROSS_COMPILE)g++
+CC := ccache $(CROSS_COMPILE)gcc
+CXX := ccache $(CROSS_COMPILE)g++
 ifeq ($(UNDER_CYGWIN),yes)
 WINDRES := $(CROSS_COMPILE)windres
 else
@@ -414,7 +414,7 @@ WINDRES_CC = $(CC)
 #>>>>> end of choices
 ###########################################################################
 
-CFLAGS = -Iproto $(DEFINES) -pipe -march=$(ARCH) -Wall
+CFLAGS = -Iproto $(DEFINES) -pipe -march=native -Wall
 CXXFLAGS = -std=gnu++11
 WINDRES_FLAGS = --preprocessor="$(WINDRES_CC) -E -xc" -DRC_INVOKED
 EXTRA_LIBS =
@@ -579,6 +579,7 @@ else
 ifeq ($(OPTIMIZE), MAXSPEED)
 CFLAGS += -O3
 CFLAGS += -fomit-frame-pointer -freg-struct-return
+LFLAGS += -Wl,-s
 else  # SPEED
 CFLAGS += -O2
 endif
@@ -785,7 +786,8 @@ endif
 
 ifeq (yes, $(USE_STDCPLUS))
 ifeq (yes, $(STATIC_STDCPLUS))
-LIB += -Wl,-Bstatic -lstdc++ -Wl,-Bdynamic
+# LIB += -Wl,-Bstatic -lstdc++ -Wl,-Bdynamic
+LIB += -static -lstdc++
 else
 LIB += -lstdc++
 endif
@@ -797,13 +799,13 @@ vimrun.exe: vimrun.c
 	$(CC) $(CFLAGS) -o vimrun.exe vimrun.c $(LIB)
 
 install.exe: dosinst.c
-	$(CC) $(CFLAGS) -o install.exe dosinst.c $(LIB) -lole32 -luuid
+	$(CC) $(CFLAGS) -o install.exe dosinst.c -lole32 -luuid $(LIB)
 
 uninstal.exe: uninstal.c
 	$(CC) $(CFLAGS) -o uninstal.exe uninstal.c $(LIB)
 
 $(TARGET): $(OUTDIR) $(OBJ)
-	$(CC) $(CFLAGS) $(LFLAGS) -o $@ $(OBJ) $(LIB) -lole32 -luuid $(LUA_LIB) $(MZSCHEME_LIBDIR) $(MZSCHEME_LIB) $(PYTHONLIB) $(PYTHON3LIB) $(RUBYLIB)
+	$(CC) $(CFLAGS) $(LFLAGS) -o $@ $(OBJ) -lole32 -luuid $(LIB) $(LUA_LIB) $(MZSCHEME_LIBDIR) $(MZSCHEME_LIB) $(PYTHONLIB) $(PYTHON3LIB) $(RUBYLIB)
 
 upx: exes
 	upx gvim.exe
@@ -906,7 +908,7 @@ ifneq (sh.exe, $(SHELL))
 	@echo 'char_u *default_vim_dir = (char_u *)"$(VIMRCLOC)";' >> pathdef.c
 	@echo 'char_u *default_vimruntime_dir = (char_u *)"$(VIMRUNTIMEDIR)";' >> pathdef.c
 	@echo 'char_u *all_cflags = (char_u *)"$(CC) $(CFLAGS)";' >> pathdef.c
-	@echo 'char_u *all_lflags = (char_u *)"$(CC) $(CFLAGS) $(LFLAGS) -o $(TARGET) $(LIB) -lole32 -luuid $(LUA_LIB) $(MZSCHEME_LIBDIR) $(MZSCHEME_LIB) $(PYTHONLIB) $(PYTHON3LIB) $(RUBYLIB)";' >> pathdef.c
+	@echo 'char_u *all_lflags = (char_u *)"$(CC) $(CFLAGS) $(LFLAGS) -o $(TARGET) -lole32 -luuid $(LIB) $(LUA_LIB) $(MZSCHEME_LIBDIR) $(MZSCHEME_LIB) $(PYTHONLIB) $(PYTHON3LIB) $(RUBYLIB)";' >> pathdef.c
 	@echo 'char_u *compiled_user = (char_u *)"$(USERNAME)";' >> pathdef.c
 	@echo 'char_u *compiled_sys = (char_u *)"$(USERDOMAIN)";' >> pathdef.c
 else
@@ -916,7 +918,7 @@ else
 	@echo char_u *default_vim_dir = (char_u *)"$(VIMRCLOC)"; >> pathdef.c
 	@echo char_u *default_vimruntime_dir = (char_u *)"$(VIMRUNTIMEDIR)"; >> pathdef.c
 	@echo char_u *all_cflags = (char_u *)"$(CC) $(CFLAGS)"; >> pathdef.c
-	@echo char_u *all_lflags = (char_u *)"$(CC) $(CFLAGS) $(LFLAGS) -o $(TARGET) $(LIB) -lole32 -luuid $(LUA_LIB) $(MZSCHEME_LIBDIR) $(MZSCHEME_LIB) $(PYTHONLIB) $(PYTHON3LIB) $(RUBYLIB)"; >> pathdef.c
+	@echo char_u *all_lflags = (char_u *)"$(CC) $(CFLAGS) $(LFLAGS) -o $(TARGET) -lole32 -luuid $(LIB) $(LUA_LIB) $(MZSCHEME_LIBDIR) $(MZSCHEME_LIB) $(PYTHONLIB) $(PYTHON3LIB) $(RUBYLIB)"; >> pathdef.c
 	@echo char_u *compiled_user = (char_u *)"$(USERNAME)"; >> pathdef.c
 	@echo char_u *compiled_sys = (char_u *)"$(USERDOMAIN)"; >> pathdef.c
 endif
diff --git a/src/Make_ming.mak b/src/Make_ming.mak
index 9ed1a70..659be93 100644
--- a/src/Make_ming.mak
+++ b/src/Make_ming.mak
@@ -11,6 +11,8 @@
 # Last updated by Ken Takata.
 # Last Change: 2014 Oct 21
 
+STATIC_STDCPLUS=yes
+USE_STDCPLUS=yes
 
 # uncomment 'PERL' if you want a perl-enabled version
 #PERL=c:/perl
@@ -33,9 +35,19 @@
 #PYTHON=/home/ron/ActivePython-2.0.0-202/src/Core
 # on my NT box, it's here:
 #PYTHON=c:/python20
+PYTHON_VER=27
+PYTHON=c:/git-sdk-64/mingw64
+PYTHON_HOME=$(PYTHON)
+PYTHONINC=-I$(PYTHON)/include/python2.7
+DYNAMIC_PYTHON_DLL=libpython2.7.dll
 
 # uncomment 'PYTHON3' if you want a python3-enabled version
 #PYTHON3=c:/python31
+PYTHON3_VER=35
+PYTHON3=c:/git-sdk-64/mingw64
+PYTHON3_HOME=$(PYTHON3)
+PYTHON3INC=-I$(PYTHON3)/include/python3.5m
+DYNAMIC_PYTHON3_DLL=libpython3.5m.dll
 
 # uncomment 'TCL' if you want a Tcl-enabled version
 #TCL=c:/tcl
