# HG changeset patch
# Parent c9781966b110ece39b8c3d58d0635bf98f2d20d1

diff --git a/src/popupmnu.c b/src/popupmnu.c
index b479b00..933d879 100644
--- a/src/popupmnu.c
+++ b/src/popupmnu.c
@@ -582,6 +582,10 @@ pum_set_selected(int n, int repeat)
 
 	    if (curwin->w_p_pvw)
 	    {
+		/* If there was already a preview window open, skip status line
+		 * redrawing */
+		if (!resized)
+		    curwin->w_redr_status = FALSE;
 		if (!resized
 			&& curbuf->b_nwindows == 1
 			&& curbuf->b_fname == NULL
