# HG changeset patch
# Parent c9781966b110ece39b8c3d58d0635bf98f2d20d1

diff -r c9781966b110 src/popupmnu.c
--- a/src/popupmnu.c	Wed Jul 16 15:44:43 2014 +0900
+++ b/src/popupmnu.c	Thu Jul 17 11:48:39 2014 +0900
@@ -574,6 +574,10 @@
 
 	    if (curwin->w_p_pvw)
 	    {
+		/* If there was already a preview window open, skip status line
+		 * redrawing */
+		if (!resized)
+		    curwin->w_redr_status = FALSE;
 		if (curbuf->b_fname == NULL
 			&& curbuf->b_p_bt[0] == 'n' && curbuf->b_p_bt[2] == 'f'
 			&& curbuf->b_p_bh[0] == 'w')
